src/audio/eq_iir/eq_iir.c

```diff
@@ -788,66 +799,45 @@ static int eq_iir_prepare(struct processing_module *mod)
 	/* EQ component will only ever have 1 source and 1 sink buffer */
 	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
 	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
-
 	source_c = buffer_acquire(sourceb);
 	sink_c = buffer_acquire(sinkb);
 
 	eq_iir_set_alignment(&source_c->stream, &sink_c->stream);
 
 	/* get source and sink data format */
+	channels = source_c->stream.channels;
 	source_format = source_c->stream.frame_fmt;
 	sink_format = sink_c->stream.frame_fmt;
 
-	source_period_bytes = audio_stream_period_bytes(&source_c->stream, dev->frames);
-	sink_period_bytes = audio_stream_period_bytes(&sink_c->stream, dev->frames);
-	if (sink_c->stream.size < sink_period_bytes) {
-		comp_err(dev, "eq_iir_prepare(): sink buffer size %d is insufficient < %d",
-			 sink_c->stream.size, sink_period_bytes);
-		buffer_release(sink_c);
-		buffer_release(source_c);
-		return -ENOMEM;
-	}
```

Comment:
Commit description doesn't mention why this stream.size check is removed? Is this related to passthrough use?

Reply:
Module adapter is doing it so no need in client. Yep, it should be in the commit message.

Reply:
Also this change needs to go to first commit.

Reply:
Done